### PR TITLE
dedicated decoder for R5G6B5_UNORM to R8G8B8A8_UNORM

### PR DESCRIPTION
**Pull request by Tillsunset**

### CODE REVIEW

**Comment by Tillsunset** [comment]
Tested with Super Mario 3d world, missing textures are good now

File: src/Cafe/HW/Latte/Core/LatteTextureLoader.h [hunk 1]
```diff
@@ -978,6 +978,57 @@ class TextureDecoder_R5_G6_B5_swappedRB : public TextureDecoder, public Singleto
 	}
 };
 
+class TextueDecoder_R5G6B5_UNORM_toR8G8B8A8_UNORM : public TextureDecoder, public SingletonClass<TextueDecoder_R5G6B5_UNORM_toR8G8B8A8_UNORM>
```
Comment (emiyl) [inline src/Cafe/HW/Latte/Core/LatteTextureLoader.h:981]: `class TextueDecoder_...`, `SingletonClass<TextueDecoder_...` typo?

File: src/Cafe/HW/Latte/Renderer/Vulkan/VulkanRenderer.cpp [hunk 2]
```diff
@@ -2642,7 +2642,7 @@ void VulkanRenderer::GetTextureFormatInfoVK(Latte::E_GX2SURFFMT format, bool isD
 		case Latte::E_GX2SURFFMT::R5_G6_B5_UNORM:
 			if (m_supportedFormatInfo.fmt_r5g6b5_unorm_pack == false) {
 				formatInfoOut->vkImageFormat = VK_FORMAT_R8G8B8A8_UNORM;
-				formatInfoOut->decoder = nullptr;
+				formatInfoOut->decoder = TextueDecoder_R5G6B5_UNORM_toR8G8B8A8_UNORM::getInstance();
```
Comment (emiyl) [inline src/Cafe/HW/Latte/Renderer/Vulkan/VulkanRenderer.cpp:2645]: And here too, `TextueDecover_...`

**Review by emiyl — CHANGES_REQUESTED** [review]

**Review by emiyl — APPROVED** [review]

**Comment by Exzap** [comment]
Looks good. Thanks!

**Pull request merged**